history/history-network.md

```diff
@@ -236,46 +236,87 @@ content          = SSZ.serialize(block_header_with_proof)
+The `find_content_ephemeral_headers_key` encodes a request for headers anchored to the block
+hash indicated by `find_content_ephemeral_headers_key.block_hash`. The
+`find_content_ephemeral_headers_key.ancestor_count` **MUST** be in the inclusive range
 0-255.
 
 The `ephemeral_header_payload` is an SSZ list of RLP encoded block header
-objects.  This object is subject to the following validity conditions.
+objects. This object is subject to the following validity conditions.
 
+* This content type **MUST** only be used for `FindContent` requests
 * The list **MAY** be empty which signals that the responding node was unable to fulfill the request.
-* The first element in the list **MUST** be the RLP encoded block header indicated by the `ephemeral_headers_key.block_hash` field from the content key.
+* The first element in the list **MUST** be the RLP encoded block header indicated by the `find_content_ephemeral_headers_key.block_hash` field from the content key.
 * Each element after the first element in the list **MUST** be the RLP encoded block header indicated by the `header.parent_hash` of the previous item from the list.
-* The list **SHOULD** contain no more than `ephemeral_headers_key.ancestor_count` items.
+* The list **SHOULD** contain no more than `find_content_ephemeral_headers_key.ancestor_count + 1` items.
+
+##### Ephemeral Header by `Offer`
+
+```python
+# Content and content key
+
+offer_ephemeral_header_key = Container(block_hash: Bytes32)
+selector                 = 0x05
+
+OfferEphemeralHeader = Container(
+  header: ByteList[MAX_HEADER_LENGTH], # RLP encoded header in SSZ ByteList
+)
+offer_ephemeral_header = OfferEphemeralHeader(header: rlp.encode(header))
+
+content          = SSZ.serialize(offer_ephemeral_header)
+content_key      = selector + SSZ.serialize(offer_ephemeral_header_key)
+```
+
+* This content type **MUST** only be used for `Offer` requests
+
+###### Ephemeral Header Offer Logic
+
+Ephemeral block headers are seeded into the network through bridges. Since ephemeral block headers are at the head of the chain, bridges should monitor if re-orgs occur. 
+
+An `Offer` message containing ephemeral headers **MUST**
+* Only contain content keys for ephemeral headers
+* Ephemeral header content keys offered to the client **MUST** be in a strictly consecutive descending order by `block_number`
+* An `Offer` message can only contain up to 31 ephemeral header content keys
+* Contain content keys from the latest tip as specified by `LightClientUpdates` to the common ancestor block if a re-org occurred. So normally this would be 1 header, unless a re-org occurred where multiple would be included. An additional `8` header's content keys, should be included in the `Offer` message as padding.
+* If bridges detect a re-org depth is larger then 31 content keys, additional `Offer` messages containing the rest of the chain should be sent by the bridge in 10 second intervals of the initial `Offer` message
```

Comment:
Should this be under a MUST block or can it be separated out as a separate note under a SHOULD or MAY under a `Details for bridges` note so it's not assumed to be required behavior for all nodes?